api/src/main/java/org/openmrs/api/handler/ConceptSaveHandler.java

```diff
@@ -57,6 +58,10 @@ public void handle(Concept concept, User creator, Date dateCreated, String other
 		}
 		if (concept.getDescriptions() != null) {
 			for (ConceptDescription cd : concept.getDescriptions()) {
+				if(cd.getLocale()==null)
+					cd.setLocale(Context.getLocale());
+				if(StringUtils.isEmpty(cd.getDescription()))
```

Comment:
Save handlers are called after validators, therefore you also need to update Concept.addDescription to not add blank descriptions so that in case no none empty description was added, it gets caught by the validator